src/components/Pagination.js

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import classnames from "classnames";
+import "../styles/Pagination.scss";
+
+const range = (from, to) => {
+  let length = to - from + 1;
+  return Array.from({ length: length }, (_, idx) => idx + from);
+}
+
+const DOTS = "...";
+
+class Pagination extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      // number of pages that gets displayed around the current page
+      siblings: 1
+    }
+  }
+  handleLeft = () => {
+    this.props.onPageChange(this.props.currentPage - 1);
+  }
+
+  handleRight = () => {
+    this.props.onPageChange(this.props.currentPage + 1);
+  }
+
+  fetchPageNumbers = () => {
+    const { currentPage, total, pageSize } = this.props;
+    const { siblings } = this.state;
+    const totalPageCount = Math.ceil(total / pageSize);
+
+    // Number of pages to display in pagination bar is determined as siblingCount + firstPage + lastPage + currentPage + 2*DOTS
```

Comment:
make sure that comments obey 80 character line limit. Break into multiple lines if it doesn't fit

Reply:
done